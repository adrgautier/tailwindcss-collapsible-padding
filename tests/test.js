const postcss = require('postcss')

it('should generate proper collapsible padding styles', () => {
  postcss([
    require('tailwindcss')({
      purge: {
        enabled: true,
        options: {
          safelist: [
            'p-px',
            'px-px',
            'py-px',
            'p-4',
            'px-4',
            'py-4',
            'sm:p-px',
            'sm:px-px',
            'sm:py-px',
            'sm:p-4',
            'sm:px-4',
            'sm:py-4',
            '2xl:p-4',
            'collapsible:p-4',
            'collapsible:py-4',
            'sm:collapsible:p-4',
            'sm:collapsible:py-4',
            'md:collapsible:py-4',
            'lg:collapsible:py-4',
            'xl:collapsible:py-4',
            '2xl:collapsible:py-4',
            'collapsible:pt-8',
            'sm:collapsible:pb-16',
            'md:collapsible:pt-32',
            'lg:collapsible:pb-64',
          ],
          variables: true,
          keyframes: true,
        },
      },
      variants: {
        extend: {
          padding: ['collapsible'],
        },
      },
      plugins: [require('../')],
    }),
  ])
    .process('@tailwind utilities', { from: undefined })
    .then(({ css }) => {
      expect(css).toMatchSnapshot()
    })
})
