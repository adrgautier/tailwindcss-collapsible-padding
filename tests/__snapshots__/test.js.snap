// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate proper collapsible padding styles 1`] = `
".p-4 {
  padding: 1rem;
}

.p-px {
  padding: 1px;
}

.collapsible\\\\:p-4::before {
  content: \\"\\";
  display: table;
  margin-bottom: 1rem;
}

.collapsible\\\\:p-4::after {
  content: \\"\\";
  display: table;
  margin-top: 1rem;
}

.collapsible\\\\:p-4 {
  padding-left: 1rem;
  padding-right: 1rem;
}

.px-4 {
  padding-left: 1rem;
  padding-right: 1rem;
}

.px-px {
  padding-left: 1px;
  padding-right: 1px;
}

.py-4 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.py-px {
  padding-top: 1px;
  padding-bottom: 1px;
}

.collapsible\\\\:py-4::before {
  content: \\"\\";
  display: table;
  margin-bottom: 1rem;
}

.collapsible\\\\:py-4::after {
  content: \\"\\";
  display: table;
  margin-top: 1rem;
}

.collapsible\\\\:pt-8::before {
  content: \\"\\";
  display: table;
  margin-bottom: 2rem;
}

*, ::before, ::after {
}

*, ::before, ::after {
}

@media (min-width: 640px) {
  .sm\\\\:p-4 {
    padding: 1rem;
  }

  .sm\\\\:p-px {
    padding: 1px;
  }

  .sm\\\\:collapsible\\\\:p-4::before {
    content: \\"\\";
    display: table;
    margin-bottom: 1rem;
  }

  .sm\\\\:collapsible\\\\:p-4::after {
    content: \\"\\";
    display: table;
    margin-top: 1rem;
  }

  .sm\\\\:collapsible\\\\:p-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }

  .sm\\\\:px-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }

  .sm\\\\:px-px {
    padding-left: 1px;
    padding-right: 1px;
  }

  .sm\\\\:py-4 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }

  .sm\\\\:py-px {
    padding-top: 1px;
    padding-bottom: 1px;
  }

  .sm\\\\:collapsible\\\\:py-4::before {
    content: \\"\\";
    display: table;
    margin-bottom: 1rem;
  }

  .sm\\\\:collapsible\\\\:py-4::after {
    content: \\"\\";
    display: table;
    margin-top: 1rem;
  }

  .sm\\\\:collapsible\\\\:pb-16::after {
    content: \\"\\";
    display: table;
    margin-top: 4rem;
  }
}

@media (min-width: 768px) {
  .md\\\\:collapsible\\\\:py-4::before {
    content: \\"\\";
    display: table;
    margin-bottom: 1rem;
  }

  .md\\\\:collapsible\\\\:py-4::after {
    content: \\"\\";
    display: table;
    margin-top: 1rem;
  }

  .md\\\\:collapsible\\\\:pt-32::before {
    content: \\"\\";
    display: table;
    margin-bottom: 8rem;
  }
}

@media (min-width: 1024px) {
  .lg\\\\:collapsible\\\\:py-4::before {
    content: \\"\\";
    display: table;
    margin-bottom: 1rem;
  }

  .lg\\\\:collapsible\\\\:py-4::after {
    content: \\"\\";
    display: table;
    margin-top: 1rem;
  }

  .lg\\\\:collapsible\\\\:pb-64::after {
    content: \\"\\";
    display: table;
    margin-top: 16rem;
  }
}

@media (min-width: 1280px) {
  .xl\\\\:collapsible\\\\:py-4::before {
    content: \\"\\";
    display: table;
    margin-bottom: 1rem;
  }

  .xl\\\\:collapsible\\\\:py-4::after {
    content: \\"\\";
    display: table;
    margin-top: 1rem;
  }
}

@media (min-width: 1536px) {
  .\\\\32xl\\\\:p-4 {
    padding: 1rem;
  }

  .\\\\32xl\\\\:collapsible\\\\:py-4::before {
    content: \\"\\";
    display: table;
    margin-bottom: 1rem;
  }

  .\\\\32xl\\\\:collapsible\\\\:py-4::after {
    content: \\"\\";
    display: table;
    margin-top: 1rem;
  }
}"
`;
